Junimatic/Patcher.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using HarmonyLib;
+using StardewModdingAPI;
+using StardewValley;
+using SObject = StardewValley.Object;
+
+namespace NermNermNerm.Junimatic
+{
+    internal static class Patcher 
+    {
+        private static ModEntry Mod = null!;
+        /// <summary>
+        /// Apply the harmony patches for Junimatic
+        /// </summary>
+        /// <param name="mod">The ModEntry instance for the Junimatic module</param>
+        /// <param name="harmony">The Harmony instance for the Junimatic module</param>
+        internal static void Apply(ModEntry mod, Harmony harmony)
+        {
+            Mod = mod;
+            harmony.Patch(
+                original: AccessTools.Method(typeof(Farmer), nameof(Farmer.gainExperience)),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Farmer_gainExperience_Prefix))
+            );
+            harmony.Patch(
+                original: AccessTools.Method(typeof(Farmer), nameof(Farmer.addItemToInventoryBool)),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Farmer_addItemToInventoryBool_Prefix))
+            );
+            harmony.Patch(
+                original: AccessTools.Method(typeof(Game1), nameof(Game1.createItemDebris)),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Game1_createItemDebris_Prefix))
+            );
+            harmony.Patch(
+                original: AccessTools.Method(typeof(Game1), nameof(Game1.createObjectDebris), [typeof(string), typeof(int), typeof(int), typeof(int), typeof(int), typeof(float), typeof(GameLocation)]),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Game1_createObjectDebris_Prefix))
+            );
+            harmony.Patch(
+                original: AccessTools.Method(typeof(Game1), "get_player"),
+                postfix: new HarmonyMethod(typeof(Patcher), nameof(Game1_get_player_Postfix))
+            );
+        }
+
+        // Track fake farmer
+        internal static Farmer Harvester = null!;
+
+        /// <summary>
+        /// Postfix to patch the calls to Game1.Player
+        /// Return FakeFarmer instead of Game1.Player when called during Junimatic harvesting logic
+        /// </summary>
+        /// <param name="__result">The FakeFarmer instance</param>
+        internal static void Game1_get_player_Postfix(ref Farmer __result) 
+        {
+            // If the fake farmer tracking variable is null, this was not called by Junimatic
+            if (Harvester is null) return;
+            __result = Harvester;
+        }
+
+        /// <summary>
+        /// Prefix to patch gainExperience in Farmer.cs
+        /// Suppress gainExperience when called during Junimatic harvesting logic.
+        /// Prevents FakeFarmer from gaining experience.
+        /// </summary>
+        /// <returns>False if called during Junimatic harvesting logic. Default: True</returns>
+        internal static bool Farmer_gainExperience_Prefix()
```

Comment:
I've been looking into this some more - I am still pretty worried about spoofing Game1.farmer.  It seems like it could cause mayhem (e.g. if our spoofed farmer doesn't have some quality of players that turns up in a later release).  But I think you need the spoofed farmer anyway, as Crop.harvest will cause the farmer to do an animation and momentarily pause for it.  It looks like there's an alternative - to pass in a spoofed JunimoHarvester, but that seems just as risky to me and doesn't carry the side-benefit of making Junimo-harvested crops less valuable and not generate xp.